ksqldb-rest-client/src/main/java/io/confluent/ksql/rest/client/KsqlClient.java

```diff
@@ -44,27 +45,50 @@
   private final HttpClient httpTlsClient;
   private final LocalProperties localProperties;
   private final Optional<String> basicAuthHeader;
+  private final Optional<HostAliasResolver> hostAliasResolver;
 
   public KsqlClient(
       final Map<String, String> clientProps,
       final Optional<BasicCredentials> credentials,
       final LocalProperties localProperties,
-      final HttpClientOptions httpClientOptions
+      final HttpClientOptions httpClientOptions,
+      final Optional<HostAliasResolver> hostAliasResolver
   ) {
     this.vertx = Vertx.vertx();
     this.basicAuthHeader = createBasicAuthHeader(
         Objects.requireNonNull(credentials, "credentials"));
     this.localProperties = Objects.requireNonNull(localProperties, "localProperties");
+    this.hostAliasResolver = Objects.requireNonNull(hostAliasResolver, "hostAliasResolver");
     this.httpNonTlsClient = createHttpClient(vertx, clientProps, httpClientOptions, false);
     this.httpTlsClient = createHttpClient(vertx, clientProps, httpClientOptions, true);
   }
 
+  public KsqlClient(
+      final Optional<BasicCredentials> credentials,
+      final LocalProperties localProperties,
+      final HttpClientOptions httpClientOptions,
+      final Consumer<HttpClientOptions> sslHttpClientOptionsConsumer,
+      final Optional<HostAliasResolver> hostAliasResolver
+  ) {
+    this.vertx = Vertx.vertx();
+    this.basicAuthHeader = createBasicAuthHeader(
+        Objects.requireNonNull(credentials, "credentials"));
+    this.localProperties = Objects.requireNonNull(localProperties, "localProperties");
+    this.hostAliasResolver = Objects.requireNonNull(hostAliasResolver, "hostAliasResolver");
+    this.httpNonTlsClient = createHttpClient(vertx, httpClientOptions, sslHttpClientOptionsConsumer,
+        false);
+    this.httpTlsClient = createHttpClient(vertx, httpClientOptions, sslHttpClientOptionsConsumer,
+        true);
+  }
+
   public KsqlTarget target(final URI server) {
     final boolean isUriTls = server.getScheme().equalsIgnoreCase("https");
     final HttpClient client = isUriTls ? httpTlsClient : httpNonTlsClient;
+    final String aliasHost = hostAliasResolver.map(resolver -> resolver.resolve(server.getHost()))
+        .orElse(server.getHost());
```

Comment:
Rather than store a `Optional<HostResolver>`, why not default to a `HostResolver` impl that just returns the value passed? e.g. `Function.identify()`.  This would simplify this code.